### ISSUE TITLE
Add Google Analytics to Create Usage Metrics

### ISSUE THREAD

**Issue by sarahjliu** [CouncilDataProject/cdp-roadmap#27]
## Idea / Feature

Incorporate Google Analytics or a similar instrumentation tool into CDP so we can track our website traffic, where people click, etc

**Comment by JacksonMaxfield** [comment]
What do you think of https://plausible.io/ ?

I have had it on councildataproject.org for the past couple of months and it works great!

Here is the report from the last month:
![image](https://user-images.githubusercontent.com/17132317/140004688-0f015f17-0b65-47ea-8c02-901e8c9f39a4.png)

(I lied, we have had 115 unique visitors which is still a lot imo!)
You can see the full stats here: https://plausible.io/councildataproject.github.io

Even with our current website being as simple as it is, people are jumping right to /#instances which may be a sign that we should deploy more instances.

**Comment by adrmac** [comment]
Hi @JacksonMaxfield @sarahjliu I am checking out the Plausible dashboard, it's ok but pretty limited in what it reports. I'm used to working with a higher level of fidelity for marketing/UX reports. It doesn't look like it supports customization. If privacy issues are a sticking point about using GA, I've heard good things about Matomo.

**Comment by adrmac** [comment]
Here's what I see in a quick scan. Most of the traffic to the site has been from Github, so presumably developers. A total of 312 unique visitors over 4 months. Nice to see some recent peaks, possibly associated with recent hackathons.
![image](https://user-images.githubusercontent.com/26727606/140419796-e2e2de71-ce47-4cb9-8d54-9ad716e8366c.png)

**Comment by JacksonMaxfield** [comment]
Some comments.

My reasons for using Plausible is:
1. I don't really want to give Google more data
2. Google Analytics would require us to put one of those annoying "cookie" alerts on the page
3. Plausible actually generally gets more correct users than GA because it isn't blocked by common adblocks / tracker blockers. I user firefox and my default settings on firefox just auto block GA
4. We can make Plausible analytics entirely public! Like with out current stats.

I largely think our data isn't "granular" enough because our website is so simple. The same setup I have for the base website would work for all the instances to get which pages and searches people are visiting. the "hash" tracker.

Take all of these stats with a caveat. Outside of a few rare instances, we have never advertised that CDP exists really. AND I am surprised we even have GA activity..... I haven't pushed any new events to the live website for seattle for months....

**Comment by JacksonMaxfield** [comment]
v3 is entirely automated while v2 is semi automated (read: semi-auto means every couple of days I need to type a command in my terminal)

**Comment by JacksonMaxfield** [comment]
oh ah plausible is entirely open source 🙂

**Comment by adrmac** [comment]
I accept your arguments. Aside from the platform used though, I think this issue thread needs to be expand to talk about what metrics are important to track and for what purpose. Roughly speaking, the main page is a marketing analysis and the instance page is a UX analysis. I'm not sure what level of customization you are able to get with Plausible, but I think we'd want event tracking for the UX side. 

I have access now to the Seattle instance GA, thanks. I'll post more thoughts as I get a chance to dig into it. At first glance it looks like we can look at data on different search terms people are querying on the app based on the URL, which is interesting.
![image](https://user-images.githubusercontent.com/26727606/140431415-9b9359ba-6758-48b5-8e91-2e98bbed3b0d.png)

**Comment by JacksonMaxfield** [comment]
> I think this issue thread needs to be expand to talk about what metrics are important to track and for what purpose.

Totally agree! Would be a great use of this thread.

With Plausible, just like with GA, we get anything that is after the `/#/` in the URL. that is how we are getting all the searches and things. So if you want something tracked all we would _really_ need to do is add something to the URL that can be deconstructed later.

It also has custom event tracking but I have never used it: https://plausible.io/docs/custom-event-goals

**Comment by sarahjliu** [comment]
Oh, the custom event tracking option on Plausible looks very helpful! I was just about to ask if we could track actions like button clicks haha

And +1 to further discussing the metrics we want to track!
The general key product metrics that come to mind are user acquisition, engagement, retention, etc. It looks like it'll be difficult to track retention/monthly active users without cookies but some initial specific metrics we could track/are already tracking include
- page views/user 
- session length/user 
- time on page for specific pages
- unique users/month for user 
- bounce rate
- top referral sources

Open to other suggestion!

Also @JacksonMaxfield what's the reasoning behind having GA for the instances and Plausible for the site? Is there a plan to merge the 2 in the future?

**Comment by JacksonMaxfield** [comment]
@sarahjliu, Seattle V2 uses GA because plausible wasn't _really_ a thing when we deployed V2.

It plan to use plausible for everything moving forward

**Comment by JacksonMaxfield** [comment]
Updating this issue: I added plausible to king county to see how it would work. Turns out because they all share the same domain under the hood, it actually is likely a really easy issue to solve.

https://plausible.io/councildataproject.github.io?period=day&date=2021-12-04

Need to update the cookiecutter to simply add the analytics script to the index.html with the some cleaned version of the "hosting_web_app_address" parameter and should be good to go. For all of our own instances, the data will be hosted in the same plausible account and we can simply create filters for the instances: https://plausible.io/councildataproject.github.io?period=7d&page=%2Fking-county%2F**
